lib/pipeline-watcher/watcher-function.ts

```diff
@@ -0,0 +1,29 @@
+import AWS = require('aws-sdk');
+
+const pipeline = new AWS.CodePipeline();
+
+/**
+ * Lambda function for checking the stages of a CodePipeline and emitting log
+ * entries with { failedCount = <no. of failed stages> } for async metric
+ * aggregation via metric filters.
+ *
+ * It requires the pipeline's name be set as the 'PIPELINE_NAME' environment variable.
+ */
+export async function handler() {
+  const pipelineName = process.env.PIPELINE_NAME;
+  if (!pipelineName) {
+    throw new Error("Pipeline name expects environment variable: 'PIPELINE_NAME'");
+  }
+
+  const state = await pipeline.getPipelineState({
+    name: pipelineName
+  }).promise();
+
+  let failedCount = 0;
+  if (state.stageStates) {
+    failedCount = state.stageStates
+      .filter(stage => stage.latestExecution !== undefined && stage.latestExecution.status === 'Failed')
+      .length;
+  }
+  process.stdout.write(JSON.stringify({failedCount}));
+}
```

Comment:
Can we write a simple unit test for this handler? (I know we don't have any UT for handlers in this library, yet, but let's raise the bar)

Reply:
Working on it .. it's a pain since we use nodeunit (deprecated) and most of what the handler does is create a client, check env variables, make a request and log to stdout - all of which are annoying to test, ha.

Reply:
See #5 - I don't think we need to use nodeunit here. Moving to jest?